src/lib/plugin/import.ts

```diff
@@ -0,0 +1,121 @@
+import { findProjectDir } from '../layout/layout'
+import { rootLogger } from '../nexus-logger'
+import { fatal } from '../process'
+import { requireModule } from '../utils'
+
+const log = rootLogger.child('plugin-manager')
+
+export interface Plugin {
+  name: string
+  worktime?: WorktimePlugin
+  runtime?: RuntimePlugin
+  testtime?: TesttimePlugin
+}
+
+/**
+ *
+ */
+export async function getInstalledRuntimePluginNames(): Promise<string[]> {
+  const packageJson = await readUsersPackageJson()
+  const pluginDepNames = extractPluginNames(packageJson)
+  const runtimePluginDepNames = pluginDepNames.filter(depName => {
+    return (
+      null !==
+      requireModule({ depName: depName + '/dist/runtime', optional: true })
+    )
+  })
+  const runtimePluginNames = runtimePluginDepNames.map(x => parsePluginName(x)!)
+  return runtimePluginNames
+}
+
+/**
+ * Load all nexus plugins installed into the project
+ */
+export async function importAllPlugins(): Promise<Plugin[]> {
+  const packageJson = await readUsersPackageJson()
+  const plugins = doImportAllPlugins(packageJson)
+  return plugins
+}
+
+/**
+ * Logic shared between sync/async variants.
+ */
+function doImportAllPlugins(packageJson: null | PackageJson): Plugin[] {
+  const pluginDepNames = extractPluginNames(packageJson)
+
+  if (!packageJson) {
+    log.trace(
+      'We could not find any package.json file. No plugin will be loaded.'
+    )
+  } else {
+    log.trace('Extracting plugins from package.json', {
+      content: packageJson,
+      pluginDepNames: pluginDepNames,
+    })
+  }
+
+  const plugins = pluginDepNames.map(depName => {
+    const pluginName = parsePluginName(depName)! // guaranteed by extract above
+    let plugin: Plugin = {
+      name: pluginName,
+    }
+    try {
+      //prettier-ignore
+      plugin.testtime = (requireModule({ depName: depName + '/dist/testtime', optional: true }) as any)?.plugin
+      //prettier-ignore
+      plugin.worktime = (requireModule({ depName: depName + '/dist/worktime', optional: true }) as any)?.plugin
+      //prettier-ignore
+      plugin.runtime = (requireModule({ depName: depName + '/dist/runtime', optional: true }) as any)?.plugin
+    } catch (error) {
+      fatal(
+        stripIndent`
+          An error occured while importing the Nexus plugin "${pluginName}":
+
+          ${error}
+        `
+      )
+    }
+
+    return plugin
+  })
+
+  return plugins
+}
+
+/**
+ * Parse a nexus plugin package name to just the plugin name.
+ */
+export function parsePluginName(packageName: string): null | string {
+  const matchResult = packageName.match(/^nexus-plugin-(.+)/)
+
+  if (matchResult === null) return null
+
+  const pluginName = matchResult[1]
+
+  return pluginName
+}
+
+/**
+ *
+ */
+async function readUsersPackageJson(): Promise<null | PackageJson> {
+  const packageJsonPath = Path.join(findProjectDir(), 'package.json')
+  return (await fs.readAsync(packageJsonPath, 'json')) ?? null
```

Comment:
future refactor:

I think layout should read and keep project package.json in memory